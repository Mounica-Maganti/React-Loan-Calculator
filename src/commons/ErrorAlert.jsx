
import React from 'react';
import PropTypes from 'prop-types';
import '../custom.css';

const ErrorAlert = ({ errorMessage }) => (
  <div className="custom-alert">{errorMessage}</div>
);

ErrorAlert.propTypes = {
  errorMessage: PropTypes.string.isRequired
};

ErrorAlert.defaultProps = {
  errorMessage: ''
};
export default ErrorAlert;