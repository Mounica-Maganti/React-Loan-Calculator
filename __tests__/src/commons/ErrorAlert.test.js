import React from 'react';
import { shallow } from 'enzyme';
import ErrorAlert from '../../../src/commons/ErrorAlert';


describe('Error Alert Tests', () => {
    let props;
    beforeEach(() => {
        props = {
            errorMessage: "Some error!!!"
        }
    });

    it('should render Loan Calculator Component', () => {
        const renderedModule = shallow(<ErrorAlert {...props} />);
        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find('.custom-alert').length).toBe(1);
        expect(renderedModule.find('.custom-alert').text()).toEqual('Some error!!!');
    });
});