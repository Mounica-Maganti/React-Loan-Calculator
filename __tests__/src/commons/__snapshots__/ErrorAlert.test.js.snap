// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Alert Tests should render Loan Calculator Component 1`] = `ShallowWrapper {}`;
